builtin/providers/aws/resource_aws_config_delivery_channel.go

```diff
@@ -0,0 +1,173 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/configservice"
+)
+
+func resourceAwsConfigDeliveryChannel() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsConfigDeliveryChannelPut,
+		Read:   resourceAwsConfigDeliveryChannelRead,
+		Update: resourceAwsConfigDeliveryChannelPut,
+		Delete: resourceAwsConfigDeliveryChannelDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "default",
+				ValidateFunc: validateMaxLength(256),
+			},
+			"s3_bucket_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
```

Comment:
Similar as above - the API docs seem don't seem to be right.

If you try creating a delivery channel w/out S3 bucket name, you'll get this error:

```
Creating Delivery Channel failed: NoSuchBucketException: Cannot find a s3 bucket with an empty bucket name.
	status code: 400, request id: 0711ca03-ee0a-11e6-b08f-2b30cd478ab2
```
and you get the same one when you make it empty string.